SYS-203 add github ci

TieDyedSheep opened the pull request:
DO NOT MERGE without approval from Andrew!

TieDyedSheep committed:
SYS-203 add github ci

muni-corn commented:
will need a separate PR to set the npm `test` script to not exit with non-zero status code :grin:

muni-corn committed:
ci: don't fail on `npm run test`

muni-corn commented:
went ahead and set the `test` script to exit with a zero status code

TieDyedSheep committed:
fixing issue comment Pr review stuff

add codeowners and sync ci.yml

fix quoting in lint script, added test and format fix (will disable test requirement for CI via repo vars)

cleanup gitlab.ci
